include/zephyr/toolchain.h

```diff
@@ -38,15 +38,15 @@
  * and therefore this header is not meant to exist in-tree
  */
 #include <toolchain/other.h>
-#elif defined(__XCC__)
+#elif defined(__XCC__) && !defined(__llvm__)
 #include <zephyr/toolchain/xcc.h>
 #elif defined(__CCAC__)
 #include <zephyr/toolchain/mwdt.h>
 #elif defined(__ARMCOMPILER_VERSION)
 #include <zephyr/toolchain/armclang.h>
-#elif defined(__llvm__) || (defined(_LINKER) && defined(__LLD_LINKER_CMD__))
+#elif (defined(__llvm__) && defined(__XCC__)) || (defined(_LINKER) && defined(__LLD_LINKER_CMD__))
```

Comment:
Both **\_\_XCC\_\_** and **\_\_llvm\_\_** are needed to identify xt-clang.

Reply:
So what happens for LLVM then?

This is not just for xt-clang.

Reply:
In the context of what consumes this header file, LLVM and xt-clang are synonymous.

The xt-clang compiler also defines \_\_XCC\_\_ , which would make it #include <zephyr/toolchain/xcc.h> instead of #include <zephyr/toolchain/llvm.h>

Reply:
> In the context of what consumes this header file, LLVM and xt-clang are synonymous.

It is not. This files is named `toolchain.h`, not `cadence-toolchain.h`.

The clang-build workflow failure below speaks for itself:

```
/__w/zephyr/zephyr/tests/lib/c_lib/common/src/main.c:144:6: error: function-like macro '__UINT8_C' is not defined
#if (UINT8_C(1) == 1)                   \
     ^
```